gnomad_qc/v4/sample_qc/relatedness.py

```diff
@@ -0,0 +1,251 @@
+            entries = entries.key_by()
+            entries = entries.select(
+                entries.row_idx, entries.col_idx, entries.n_alt_alleles
+            )
+
+            # Export one compressed Parquet file per partition.
+            logger.info("Writing Parquet tables...")
+            entries.to_spark().write.option("compression", "zstd").parquet(input_path)
+
+            # Write metadata that's useful for postprocessing. Map `col_idx` to `s`
+            # explicitly so we don't need to rely on a particular order returned by
+            # `collect`.
+            logger.info("Writing metadata...")
+            col_idx_mapping = hl.struct(col_idx=mt.col_idx, s=mt.s).collect()
+            col_idx_mapping.sort(key=lambda e: e.col_idx)
+            metadata = {
+                "num_sites": row_count_v4,
+                "samples": [e.s for e in col_idx_mapping],
+            }
+            with hadoop_open(f"{input_path}/metadata.json", 'w') as f:
+                json.dump(metadata, f)
+
+        if args.create_relatedness_table:
+            from hail.utils.java import Env
+
+            spark = Env.spark_session()
+            df = spark.read.parquet(cuking_output_path(test=test))
+            ht = hl.Table.from_spark(df)
+            ht = ht.repartition(64)
+            ht = ht.key_by(ht.i, ht.j)
+            ht.write(relatedness.path)
+
+        if args.compute_related_samples_to_drop:
+            # compute_related_samples_to_drop uses a rank table as a tie breaker when
+            # pruning samples. We determine rankings as follows: filtered samples are
+            # ranked lowest, followed by non-releasable samples, followed by presence in
+            # v3 (favoring v3 over v4), and then by coverage.
+            def annotate_project_meta_ht(
+                project_meta_ht, sex_ht, hard_filtered_samples_ht
+            ):
+                return project_meta_ht.select(
+                    project_meta_ht.releasable,
+                    project_meta_ht.exclude,
+                    chr20_mean_dp=sex_ht[project_meta_ht.key].chr20_mean_dp,
+                    filtered=hl.or_else(
+                        hl.len(
+                            hard_filtered_samples_ht[project_meta_ht.key].hard_filters
+                        )
+                        > 0,
+                        False,
+                    ),
+                )
+
+            from gnomad_qc.v3.resources.meta import project_meta as project_meta_v3
+            from gnomad_qc.v4.resources.meta import project_meta as project_meta_v4
+            from gnomad_qc.v3.resources.sample_qc import (
+                hard_filtered_samples as hard_filtered_samples_v3,
+            )
+            from gnomad_qc.v4.resources.sample_qc import (
+                hard_filtered_samples as hard_filtered_samples_v4,
+            )
+            from gnomad_qc.v3.resources.sample_qc import sex as sex_v3
+            from gnomad_qc.v4.resources.sample_qc import sex as sex_v4
```

Comment:
Why here instead of at the beginning of the file?

Reply:
Just because of the v3 + v4 renaming, which isn't used anywhere else in the code. But happy to move that to the top (if still required).

Reply:
No longer needed with the new PR I just put in